haskell/cc.bzl

```diff
@@ -112,9 +113,6 @@ def cc_interop_info(ctx):
     cc_files = ctx.files._cc_toolchain + cc_wrapper.inputs.to_list()
     cc_manifests = cc_wrapper.manifests
 
-    # XXX Workaround https://github.com/bazelbuild/bazel/issues/6876.
-    linker_flags = [flag for flag in linker_flags if flag not in ["-shared"]]
```

Comment:
While debugging #1349 I noticed that this could be simplified by following the suggestion from https://github.com/bazelbuild/bazel/issues/6876#issuecomment-452727209.